test/csuite/wt8659_reconstruct_database_from_logs/main.c

```diff
@@ -0,0 +1,330 @@
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+ * OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+#include <test_util.h>
+
+#define MAX_ITERATIONS 5
+#define MAX_KEYS 100000
+
+static const char *const conn_config =
+  "create,cache_size=100MB,log=(archive=false,enabled=true,file_max=100K)";
+static const char *const home_live = "WT_HOME_LOG";
+static const char *const home_full = "WT_HOME_LOG_FULL";
+static const char *const home_incr = "WT_HOME_LOG_INCR";
+static const char *const home_incr_copy = "WT_HOME_LOG_INCR_COPY";
+static const char *const uri = "table:logtest";
+static const char *const full_out = "./backup_full";
+static const char *const incr_out = "./backup_incr";
+
+static WT_CONNECTION *conn = NULL;
+static WT_SESSION *session = NULL;
+
+/*
+ * dump_table --
+ *     Dumps the table content in to the file in human-readable format.
+ */
+static void
+dump_table(const char *home, const char *table, const char *out_file)
+{
+    char buf[1024];
+
+    testutil_check(
+      __wt_snprintf(buf, sizeof(buf), "./wt -R -h %s dump %s > %s", home, table, out_file));
+    testutil_check(system(buf));
+}
+
+/*
+ * reset_dir --
+ *     Recreates the directory.
+ */
+static void
+reset_dir(const char *dir)
+{
+    char buf[1024];
+
+    testutil_check(__wt_snprintf(buf, sizeof(buf), "rm -rf %s && mkdir %s", dir, dir));
+    testutil_check(system(buf));
+}
+
+/*
+ * remove_dir --
+ *     Removes the directory.
+ */
+static void
+remove_dir(const char *dir)
+{
+    char buf[1024];
+
+    testutil_check(__wt_snprintf(buf, sizeof(buf), "rm -rf %s", dir));
+    testutil_check(system(buf));
+}
+
+/*
+ * compare_backups --
+ *     Compares the full and the incremental backups.
+ */
+static int
+compare_backups(void)
+{
+    int ret;
+    char buf[1024];
+
+    /*
+     * We have to copy incremental backup to keep the original database intact. Otherwise we'll get
+     * "Incremental backup after running recovery is not allowed".
+     */
+    testutil_check(__wt_snprintf(buf, sizeof(buf), "cp %s/* %s", home_incr, home_incr_copy));
+    testutil_check(system(buf));
+
+    /* Dump both backups. */
+    dump_table(home_full, uri, full_out);
+    dump_table(home_incr_copy, uri, incr_out);
+
+    reset_dir(home_incr_copy);
+
+    /* Compare the files. */
+    testutil_check(__wt_snprintf(buf, sizeof(buf), "cmp %s %s", full_out, incr_out));
+    if ((ret = system(buf)) != 0) {
+        printf(
+          "Tables \"%s\" don't match in \"%s\" and \"%s\"!\n See \"%s\" and \"%s\" for details.\n",
+          uri, home_full, home_incr_copy, full_out, incr_out);
+        exit(1);
```

Comment:
Should we do instead `exit (ret)`? Not sure what will be the value of `ret` if the comparison fails though.

Reply:
`cmp` returns 
 - 0 – files are identical
 - 1 – files differ
 - 2 – inaccessible or missing argument

C program return codes
 - 0 (EXIT_SUCCESS): success
 - 1 (EXIT_FAILURE): It indicates abnormal termination of a program perhaps as a result of a minor problem in the code.
 - 2: It is similar to EXIT_FAILURE but is displayed when the error occurred is a major one. This statement is rarely seen.

I think we should stick to the generic exit(1). What do you think?

Reply:
Sounds good, let's keep 1.